### PR TITLE
Adds ys_prune

### PR DESCRIPTION
**Pull request by kylebaron**
# Summary
- Select any column from a data frame that appears in the `spec` object
- Column order of the result is same as order in `spec`
- Optionally report columns in `spec` that aren't in `data`
- Error if no columns are in common between `data` and `spec`

### CODE REVIEW

File: R/class-yspec.R [hunk 1]
```diff
@@ -429,6 +429,56 @@ ys_add_labels <- function(data,spec,fun=label.ycol) {
...
 
+
+#' Prune a data frame, keeping columns in a yspec object
+#' 
+#' Use this to scavenge a data frame for columns that you want to keep. Do not
+#' use this for final column selection; use [dplyr::select()] instead. 
+#' 
+#' @param data a data frame with at least one column that is found in `spec`
+#' @param spec a `yspec` object
+#' @param report if `TRUE`, report mising columns
```
Comment (andersone1) [inline R/class-yspec.R:440]: typo - "mising"

File: tests/testthat/test-prune.R [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+library(testthat)
+
+context("text-prune")
+
+test_that("ys_prune selects available columns", {
+  data <- ys_help$data()
+  spec <- ys_help$spec()
+  data$STUDY <- NULL
+  data$TAD <- NULL
+  ans <- ys_prune(data, spec)
```
Comment (andersone1) [inline tests/testthat/test-prune.R:11]: @kylebaron 

should this test have a data frame with more columns than the spec?

For example

- data has columns: A, B, C
- spec has columns: A, B

The current test has the reverse, and therefore:

`names(ys_prune(data, spec)) == names(data)`

**Review by andersone1 — CHANGES_REQUESTED** [review]

File: R/class-yspec.R (hunk 1 above)
Comment (kylebaron) [reply R/class-yspec.R:440]: Fixed.

File: tests/testthat/test-prune.R (hunk 2 above)
Comment (kylebaron) [reply tests/testthat/test-prune.R:11]: Yes; I was intending fewer columns. Can add more columns too.

Comment (kylebaron) [reply tests/testthat/test-prune.R:13]: Added two columns matching nothing.

Comment (andersone1) [reply tests/testthat/test-prune.R:13]: @kylebaron 

One other question - if I add:

`data <- data[, rev(names(data)), drop = FALSE]`

after line 12 (`data$BAR <- 2`) but before line 13 (`ans <- ys_prune(data, spec))`), the test then fails.

Is this OK?

Comment (kylebaron) [reply tests/testthat/test-prune.R:15]: Thanks, @andersone1 ; yes, there was something wrong in the logic for matching / ordering the names.  

- Fixed the logic in `ys_prune`
- Randomized the names in `data` in the test
- Added expectations for reporting columns not there

Thanks!

Comment (kylebaron) [reply tests/testthat/test-prune.R:15]: I reverted that one expectation; I do think we want this to be the check:

```r
  expect_equal(names(ans), names(spec2))
```

The names in the final answer are identical (including order) to the names in the spec, minus the columns that we dropped.

**Review by andersone1 — APPROVED** [review]

**Pull request merged**